### PR TITLE
reuse shared gha action

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: Olivier Lamy <olamy@apache.org>

### CODE REVIEW

**Review by joakime — APPROVED** [review]

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -26,26 +26,7 @@ on:
...
-        uses: actions/setup-java@v3
-        with:
-          distribution: ${{ matrix.jdk }}
-          java-version: ${{ matrix.java }}
-          cache: 'maven'
-
-      - name: Build with Maven
-        run: mvn install javadoc:javadoc -e -B -V
+    name: Maven Build CI
+    uses: jetty/.github/.github/workflows/maven-ci.yml@main
```
Comment (joakime) [inline .github/workflows/maven.yml:30]: Usage of `@main` means there's no caching.
The advice of github is to use a version, specifically from a tag.
Aka `jetty/.github/.github/workflows/maven-ci.yml@5`

**Review by joakime — CHANGES_REQUESTED** [review]

File: .github/workflows/maven.yml (hunk 1 above)
Comment (joakime) [reply .github/workflows/maven.yml:30]: Using the branch name is really only appropriate when you are testing things.

Comment (olamy) [reply .github/workflows/maven.yml:30]: > Usage of `@main` means there's no caching. The advice of github is to use a version, specifically from a tag. Aka `jetty/.github/.github/workflows/maven-ci.yml@5`

Which caching are referring to? the most important is this cache which is based on pom.xml hash (https://github.com/actions/setup-java#caching-packages-dependencies)

The security advice applies for random actions you are using, And the only immutable would be using a git hash (tags are not immutable)
Here we use a branch of your own github action, we do not inject any token so the risk is almost null.
Maybe the shared action itself code use tag (but even tags is not really secured)
And by the way we are using only actions from this https://github.com/actions (which is owned by Github) so if Github repo is compromised we have definitely more problems to solve)

Comment (olamy) [reply .github/workflows/maven.yml:30]: > Using the branch name is really only appropriate when you are testing things.

this was already the case when using this
```
      - name: Checkout
        uses: actions/checkout@v4

      - name: Set up JDK
        uses: actions/setup-java@v3
```
At least here it's our own code so we have more control.
Anyway using tag/version of our own code sounds good.

**Comment by olamy** [comment]
let's forget that! copy/paste is the modern tooling for 2024

**Pull request closed without merging**